optuna/samplers/nsgaii/_elite_population_selection_strategy.py

```diff
@@ -109,42 +112,30 @@ def _crowding_distance_sort(population: list[FrozenTrial]) -> None:
     population.reverse()
 
 
-def _fast_non_dominated_sort(
+def _rank_population(
     population: list[FrozenTrial],
-    directions: list[optuna.study.StudyDirection],
-    dominates: Callable[[FrozenTrial, FrozenTrial, list[optuna.study.StudyDirection]], bool],
+    directions: Sequence[StudyDirection],
+    *,
+    is_constrained: bool = False,
 ) -> list[list[FrozenTrial]]:
-    dominated_count: defaultdict[int, int] = defaultdict(int)
-    dominates_list = defaultdict(list)
-
-    for p, q in itertools.combinations(population, 2):
-        if dominates(p, q, directions):
-            dominates_list[p.number].append(q.number)
-            dominated_count[q.number] += 1
-        elif dominates(q, p, directions):
-            dominates_list[q.number].append(p.number)
-            dominated_count[p.number] += 1
-
-    population_per_rank = []
-    while population:
-        non_dominated_population = []
-        i = 0
-        while i < len(population):
-            if dominated_count[population[i].number] == 0:
-                individual = population[i]
-                if i == len(population) - 1:
-                    population.pop()
-                else:
-                    population[i] = population.pop()
-                non_dominated_population.append(individual)
-            else:
-                i += 1
-
-        for x in non_dominated_population:
-            for y in dominates_list[x.number]:
-                dominated_count[y] -= 1
-
-        assert non_dominated_population
-        population_per_rank.append(non_dominated_population)
+    if len(population) == 0:
+        return []
+
+    objective_values = np.array(
+        [
+            trial.values if trial.values else [float("inf")] * len(directions)
+            for trial in population
+        ],
+        dtype=np.float64,
+    )
+    objective_values *= np.array(
```

Comment:
Can we use `loss_values` to be clear to us in the future that each objective is better when it is lower in this array?
Please check out other functions as well.

Reply:
Just to give @HideakiImamura the context:
`loss_vals` (or another name is `lvals`) is often used in `TPESampler` (please check [here](https://github.com/optuna/optuna/blob/master/optuna/samplers/_tpe/sampler.py#L610-L622) and [here](https://github.com/optuna/optuna/blob/master/optuna/samplers/_tpe/sampler.py#L687-L735))

So the problem here is that I would need to take a look at the source of the function calls whether `objective_values` is always better when each objective is lower.
loss does not necessarily mean the machine-learning loss functions, but just loss, whichwe already have a universal consensus such that it is better when loss is lower even in normal conversations.

I do not mind using `objective_values`, but I strongly encourage you to specify whether each objective is better when it is lower.
Again, the reason is simple because I would need to refer to the function call origin to see if `objective_values` is always better when it is lower.